vtm/src/org/oscim/core/MapElement.java

```diff
@@ -75,11 +75,18 @@ public String toString() {
     public MapElement clone() {
         MapElement copy = new MapElement();
         copy.tags.set(this.tags.asArray());
-        copy.points = Arrays.copyOf(this.points, this.points.length);
+        copy.points = Arrays.copyOf(this.points, this.pointPos);
         copy.pointPos = this.pointPos;
         copy.labelPosition = this.labelPosition;
         copy.setLayer(this.layer);
-        copy.index = Arrays.copyOf(this.index, this.index.length);
+        int indexSize = this.indexPos + 1;
+        for (int i = 0; i < this.index.length; i++) {
+            if (index[i] == -1) {
+                indexSize = i + 1;
```

Comment:
Shouldn't be `indexSize = i;`?
Since the `i` is empty (`-1`), it declares also the size (with `0` start)?

Reply:
Thought maybe some readers need the `-1` value to finish reading array (e.g. `while(index[i] != -1){i++}`, but not seen yet); 
But the size itself can act as regular finishing argument 👍